src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -719,4 +724,24 @@ private void validateStaticCodeAnalysisSettings(ProgrammingExercise programmingE
         programmingExerciseService.deleteTasksWithSolutionEntries(exercise.getId());
         return ResponseEntity.noContent().build();
     }
+
+    /**
+     * Returns the solution repository files with content for a given programming exercise.
+     * Note: This endpoint redirects the request to the ProgrammingExerciseParticipationService. This is required if
+     * the solution participation id is not known for the client.
+     * @param exerciseId the exercise for which the solution repository files should be retrieved
+     * @return a redirect to the endpoint returning the files with content
+     */
+    @GetMapping(SOLUTION_REPOSITORY_FILES_WITH_CONTENT)
+    @PreAuthorize("hasRole('TA')")
+    @FeatureToggle(Feature.ProgrammingExercises)
+    public ModelAndView redirectGetSolutionRepositoryFiles(@PathVariable Long exerciseId) {
```

Comment:
We could let the client handle this. But I think this endpoint could also be interesting for other scenarios.

Reply:
Could you maybe elaborate a little on what scenarios would benefit from this endpoint?

Reply:
Idk, it seems to me that it could be useful in the feature.

In addition, the only service accessing this endpoint is the `code-editor-repository.service.ts`. But using this endpoint makes the code way more complex for no reason and does not really fit the purpose of the `programming-exercise-detail-component`. For example, you have to set a domain to the service. It may be helpful if you can a look at that service yourself.
So the other option would be to implement a new ts-service for accessing the SolutionProgrammingExerciseParticipationResource. Not sure if that's really a better idea.

Reply:
Alright. From my side, you can keep it like this then.